### PR TITLE
Handle OOM errors by stopping the periodic log collector

### PR DESCRIPTION
**Pull request by kevinclark19a**
## Description

A bug exists in the current code base where specific circumstances lead to the periodic log collector getting consistently OOM killed. This seems to not be recoverable in the current code base, so this PR introduces logic to turn off the log collector (until service restart) the first time this occurs.

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: azurelinuxagent/common/logcollector.py [hunk 1]
```diff
@@ -45,6 +45,7 @@
 
 CGROUPS_UNIT = "collect-logs.scope"
 
+OOM_KILLED_ERRCODE = -9
```
Comment (narrieta) [inline azurelinuxagent/common/logcollector.py:48]: I would not make this specific to OOM... -9 it is just a generic force kill. let's rename this and add comments that it being used to attempt to capture the termination by OOM

Comment (kevinclark19a) [reply azurelinuxagent/common/logcollector.py]: Updated. I added the comments where this value is used, thought that fit better.

**Review by narrieta — APPROVED** [review]

File: azurelinuxagent/ga/collect_logs.py [hunk 2]
```diff
@@ -203,6 +203,12 @@ def exec_command(output_file):
                     if e.returncode == logcollector.INVALID_CGROUPS_ERRCODE: # pylint: disable=no-member
                         logger.info("Disabling periodic log collection until service restart due to process error.")
                         self.stop()
+                    
+                    # When the OOM killer is invoked on the log collector process, this error code is
+                    # returned. Stop the periodic operation because it seems to be persistent.
+                    elif e.returncode == logcollector.FORCE_KILLED_ERRCODE: # pylint: disable=no-member
+                        logger.info("Disabling periodic log collection until service restart due OOM error.")
```
Comment (nagworld9) [inline azurelinuxagent/ga/collect_logs.py:210]: "service restart due _to_ OOM error"?

**Review by nagworld9 — DISMISSED** [review]

File: azurelinuxagent/ga/collect_logs.py (hunk 2 above)
Comment (kevinclark19a) [reply azurelinuxagent/ga/collect_logs.py:210]: Yep, good catch

**Review by nagworld9 — APPROVED** [review]

**Pull request merged**